Kernel/VirtIO/VirtIOConsolePort.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ * Copyright (c) 2021, Kyle Pereira <hey@xylepereira.me>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <Kernel/Devices/CharacterDevice.h>
+#include <Kernel/FileSystem/FileDescription.h>
+#include <Kernel/VM/RingBuffer.h>
+#include <Kernel/VirtIO/VirtIO.h>
+
+namespace Kernel {
+
+class VirtIOConsole;
+
+#define VIRTIO_CONSOLE_F_SIZE (1 << 0)
+#define VIRTIO_CONSOLE_F_MULTIPORT (1 << 1)
+#define VIRTIO_CONSOLE_F_EMERG_WRITE (1 << 2)
+
+class VirtIOConsolePort
+    : public CharacterDevice
+    , RefCounted<VirtIOConsolePort> {
```

Comment:
why is this refcounted?

Reply:
Err don't I have to make it RefCounted so that VirtIOConsole can have an OwnPtr to it?

Reply:
no, RefCounted allows you to hold a RefPtr to it :)

Reply:
Huh, so then what do I need to hold an OwnPtr? It doesn't compile if I remove that

Reply:
Nothing, OwnPtr can hold any type, whats the error?

Reply:
Uh it says this:

```/.././AK/NonnullOwnPtr.h: In instantiation of 'AK::NonnullOwnPtr<T>::NonnullOwnPtr(AK::NonnullOwnPtr<T>::AdoptTag, T&) [with T = Kernel::VirtIOConsolePort]':
../.././AK/NonnullOwnPtr.h:160:12:   required from 'AK::NonnullOwnPtr<T> AK::make(Args&& ...) [with T = Kernel::VirtIOConsolePort; Args = {unsigned int, Kernel::VirtIOConsole&}]'
../../Kernel/VirtIO/VirtIOConsole.cpp:49:55:   required from here
../.././AK/NonnullOwnPtr.h:36:64: error: static assertion failed: Use NonnullRefPtr<> for RefCounted types
   36 |             requires { requires typename T::AllowOwnPtr()(); } || !requires(T obj) { requires !typename T::AllowOwnPtr()(); obj.ref(); obj.unref(); },
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../.././AK/NonnullOwnPtr.h:36:64: note: '(false || false)' evaluates to false
```
Upon reading it now I think I can also say `using AllowOwnPtr = TrueType;`? There's only one other place where that's used so I dunno if that's right?

Reply:
Well theres 2 separate issues there, firstly the VirtIOConsole doesnt have to be refcounted anyways even if OwnPtr did require that, since you dont take a VirtIOConsole in an OwnPtr anywhere.
the issue with owning a VirtIOConsolePort is described in that error message if you read it, you are not allowed to use `OwnPtr` on refcounted types, and VirtIOConsolePort inherits from `File`, which is refcounted, the correct solution here is to just hold a `RefPtr` to the VirtIOConsolePort and thats it.

Reply:
Huh, that simple? Guess I should've just read it lol. 

Oh and the RefCounted on VirtIOConsole was unintentional, just something I added and forgot to remove.